### PR TITLE
Bump cw-orch and use `MockBech32`

### PR DESCRIPTION
**Pull request by CyberHoward**
Bumps cw-orch and uses bech32 mock because of instantiate2 use. Replaces #239.

Migrate tests get a fix here : https://github.com/AbstractSDK/abstract/pull/256

## Checklist

- [x] Finish interchain tests

- [x] Replace cw-orch patch by published version when bech32 mock is released : https://github.com/AbstractSDK/cw-orchestrator/pull/325
- In framework
- In modules
- In integrations
- In interchain

- [x] Replace cw-orch-interchain by tagged version when bech32 fix is released : https://github.com/AbstractSDK/cw-orch-interchain/pull/28
- In interchain

### CODE REVIEW

File: modules/contracts/apps/croncat/tests/integration.rs [hunk 1]
```diff
@@ -74,6 +70,8 @@ fn setup_croncat_contracts(
         None,
     )?;
 
+    println!("{:?}", cw20_addr);
+
```
Comment (Buckram123) [inline modules/contracts/apps/croncat/tests/integration.rs:74]: Not sure if it should belong here?

File: modules/contracts/apps/challenge/tests/integrations.rs [hunk 2]
```diff
@@ -31,63 +30,87 @@ const FIRST_CHALLENGE_ID: u64 = 1;
 
 const INITIAL_BALANCE: u128 = 50_000_000;
 
-lazy_static! {
-    static ref CHALLENGE_REQ: ChallengeRequest = ChallengeRequest {
+lazy_static! {}
```
Comment (Buckram123) [inline modules/contracts/apps/challenge/tests/integrations.rs:33]: It's empty lazy static, remove?

File: modules/contracts/apps/calendar/tests/integration.rs [hunk 3]
```diff
@@ -83,49 +83,61 @@ fn setup_with_time(
...
+        .asset(DENOM, AssetInfoUnchecked::native(DENOM))
+        .build()?;
 
     client.set_balances(vec![
-        ("sender1", coins(INITIAL_BALANCE, DENOM).as_slice()),
-        ("sender2", coins(INITIAL_BALANCE, DENOM).as_slice()),
-        ("sender", coins(INITIAL_BALANCE, DENOM).as_slice()),
+        (
+            chain.create_account("sender1"),
+            coins(INITIAL_BALANCE, DENOM).as_slice(),
```
Comment (Buckram123) [inline modules/contracts/apps/calendar/tests/integration.rs:98]: Feels like we want to have `chain.create_account_with_balance()` here?

**Review by Buckram123 — COMMENTED** [review]
This change looks VERY painful to do, noticed small nits

File: app-template/tests/integration.rs [hunk 4]
```diff
@@ -22,11 +22,12 @@ fn setup(
     count: i32,
 ) -> anyhow::Result<(AbstractClient<Mock>, Application<Mock, AppInterface<Mock>>)> {
     // Create a sender
-    let sender = Addr::unchecked(OWNER);
+    let mock = Mock::new("mock");
+    let sender = mock.create_account("sender");
```
[the remark this answers is not in the document]
Comment (CyberHoward) [reply app-template/tests/integration.rs:26]: what do you mean?

[the remark this answers is not in the document]
Comment (CyberHoward) [reply app-template/tests/integration.rs:26]: You mean he should use `.sender()` to get the addr?

[the remark this answers is not in the document]
Comment (Buckram123) [reply app-template/tests/integration.rs:26]: > You mean he should use `.sender()` to get the addr?

Yeah, I mean we create "mock" as sender but never actually use it here

File: modules/contracts/apps/calendar/tests/integration.rs (hunk 3 above)
Comment (Kayanski) [reply modules/contracts/apps/calendar/tests/integration.rs]: Like so : https://github.com/AbstractSDK/cw-orchestrator/pull/332 ?

Comment (Buckram123) [reply modules/contracts/apps/calendar/tests/integration.rs]: Yeah, nice!

**Pull request merged**